project/src/main/webapp/script.js

```diff
@@ -13,6 +13,32 @@
 // limitations under the License.
 
 window.onload = function onLoad() {
+  checkLoginStatus();
+};
+
+/**
+ * Checks login status and displays navbar and profile if user is logged in,
+ * and redirects to index page if user is not logged in.
+ */
+function checkLoginStatus() {
+  fetch('/login-status')
+      .then((response) => {
+        return response.json();
+      })
+      .then((loginStatus) => {
+        const isLoggedIn = loginStatus.isLoggedIn;
+        if (isLoggedIn) {
+          loadHeader();
+        } else {
+          window.href = 'index.html';
+        }
```

Comment:
Same here. e.g.

then(response) => {
  const loginStatus = respone.json;
...
}

Reply:
Tried this but I think it does not work because response.json() returns a promise since response is a stream.

Reply:
Have you tried like const loginStatis = await respone.json(); This can be cleaner.

Reply:
Done - thank you, it is cleaner now.